x/concentrated-liquidity/pool.go

```diff
@@ -101,6 +101,41 @@ func (k Keeper) setPool(ctx sdk.Context, pool types.ConcentratedPoolExtension) e
 	return nil
 }
 
+func (k Keeper) GetPoolDenoms(ctx sdk.Context, poolId uint64) ([]string, error) {
+	concentratedPool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return nil, err
+	}
+
+	denoms := []string{concentratedPool.GetToken0(), concentratedPool.GetToken1()}
+	return denoms, nil
+}
+
+func (k Keeper) CalculateSpotPrice(
+	ctx sdk.Context,
+	poolId uint64,
+	quoteAssetDenom string,
+	baseAssetDenom string,
+) (spotPrice sdk.Dec, err error) {
+	concentratedPool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+
+	sqrtPrice := concentratedPool.GetCurrentSqrtPrice().Power(2)
+	// if sqrt price is zero, it means that two assets are with same ratio
+	// this we state that spot price is 1
+	if sqrtPrice.IsZero() {
+		return sdk.OneDec(), nil
+	}
```

Comment:
This logic does not follow for me. Sqrt price of zero should not be possible, and even if it was, a sqrtPrice of zero should not equal a spot price of 1.

Reply:
See comment here https://github.com/osmosis-labs/osmosis/pull/4299#discussion_r1105982854

Reply:
See comment here https://github.com/osmosis-labs/osmosis/pull/4299#discussion_r1107408872